astropy/convolution/setup_package.py

```diff
@@ -3,9 +3,8 @@
 import os
 import sys
 
-from setuptools import Extension
```

Comment:
Doesn't it make more sense to import setuptools before you import numpy for this case?

Reply:
IDK, why?

Reply:
setuptools is from Python org and numpy is our own third-party dep. I think the existing order makes more sense.

Reply:
Oh. I don't think isort distinguishes between third-party dependencies. We could add a category between `STDLIB` and `THIRD-PARTY`, like `INFRASTRUCTURE`...

Reply:
Though I'd like to get more discussion before making that more drastic change. @mhvk @WilliamJamieson @eerovaher @saimn (just some people recently in these formatting discussions).

Reply:
I realize I thought `setuptools` was in the standard library, which it is not...  My sense would be to just let `isort` do its work even though @pllim's comment feels right, that it should be earlier.

Reply:
We have a ``build_requires`` in the ``pyproject.toml``, which feels like a subcategory of third-party dependencies, so  I'm either way on separating these out.

Reply:
Vote tally
- keep.   :  +1 @mhvk, +.5 @nstarman 
- change: +1 @pllim,  +.5 @nstarman

Reply:
My vote is for the simplest isort configuration possible.

Reply:
Really more like +0.3 or so... But simplest configuration possible is attractive.

Reply:
@pllim how about we leave isort as is for now and open an issue to discuss future refinement? With @WilliamJamieson's PR to auto run pre-commit any decisions will propagate to the whole repo 🥳

Reply:
#13718 if we want to discuss before/during/after this PR is merged.

Reply:
Sure, CI passed after all, so not like the order really matters; just a human preference.